src/test/migrate.test.ts

```diff
@@ -0,0 +1,92 @@
+import {describe, it, expect, beforeAll, afterAll} from 'vitest';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import {
+  createTempDir,
+  cleanupTempDir,
+  runCliProcess,
+  stripVersion
+} from './utils.js';
+
+let tempDir: string;
+
+beforeAll(async () => {
```

Comment:
we should do this inside the `describe` as a `beforeEach` and `afterEach` - so every individual test has a clean fixture